### ISSUE TITLE
Supported Layers Tracker

### ISSUE THREAD

**Issue by mohmdelsayed** [mohmdelsayed/HesScale#1]
Here is a list of the `torch.nn` modules to be supported for HesScale. 

- [x] **Loss functions**
  - [x] MSELoss
  - [x] CrossEntropyLoss
- [x] **Activations**
  - [x] ReLU
  - [x] Sigmoid
  - [x] Tanh
  - [x] LeakyReLU
  - [x] LogSigmoid
  - [x] ELU
  - [x] SELU
- [x] Others:
  - [x] Linear
  - [x] Dropout
  - [x] Flatten
- [ ] **CNN related layers**:
  - [x] MaxPool1d
  - [x] MaxPool2d
  - [x]  AvgPool1d
  - [x] MaxPool3d
  - [x] AvgPool2d
  - [x] AvgPool3d
  - [ ] ZeroPad2d
  - [x] Conv1d
  - [x] Conv2d
  - [x] Conv3d
  - [x] ConvTranspose1d
  - [x] ConvTranspose2d
  - [x] ConvTranspose3d
 - [ ] **Recurrent related layers**:
    - [ ] RNN
    - [ ] LSTM
